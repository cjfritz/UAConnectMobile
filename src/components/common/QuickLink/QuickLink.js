import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {
  Text,
} from 'native-base';
import styles from './QuickLink.style';

const QuickLink = ({
  onPress, children, color, iconName,
}) => (
  <TouchableOpacity onPress={ onPress } style={ { ...styles.Touch, backgroundColor: color } }>
    <Icon name={ iconName } style={ styles.Icon } size={ 100 } />
    <Text>
      { children }
    </Text>
  </TouchableOpacity>
);

export default QuickLink;

