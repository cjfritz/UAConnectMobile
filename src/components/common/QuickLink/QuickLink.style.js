import { StyleSheet } from 'react-native'; 

export default StyleSheet.create({
  Icon: {
    color: 'black',
  },
  Touch: {
    alignItems: 'center',
    borderBottomLeftRadius: 20,
    borderBottomRightRadius: 20,
    borderBottomWidth: 5,
    borderTopEndRadius: 20,
    borderTopStartRadius: 20,
    flex: 1,
    justifyContent: 'center',
    marginBottom: 40,
    marginLeft: 20,
    marginRight: 20,
    marginTop: 20,
  },
});