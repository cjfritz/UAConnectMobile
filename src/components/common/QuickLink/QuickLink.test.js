import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import QuickLink from './QuickLink';

const props = {
  onPress: jest.fn(),
  children: 'Profile',
  color: 'white',
  iconName: 'ios-person',
};

describe('<QuickLink />', () => {
  it('renders with profile props', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<QuickLink { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with course plannner props', () => {
    const newProps = {
      children: 'Course Planner',
      iconName: 'ios-bookmarks',
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<QuickLink { ...props } { ...newProps } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with event props', () => {
    const newProps = {
      children: 'Event',
      iconName: 'ios-calendar',
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<QuickLink { ...props } { ...newProps } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with news props', () => {
    const newProps = {
      children: 'News',
      iconName: 'md-paper',
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<QuickLink { ...props } { ...newProps } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
