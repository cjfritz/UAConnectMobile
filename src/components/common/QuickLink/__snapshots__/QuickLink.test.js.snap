// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickLink /> renders with course plannner props 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 5,
      "borderTopEndRadius": 20,
      "borderTopStartRadius": 20,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 40,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 20,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="ios-bookmarks"
    size={100}
    style={
      Object {
        "color": "black",
      }
    }
  />
  <Styled(Text)>
    Course Planner
  </Styled(Text)>
</TouchableOpacity>
`;

exports[`<QuickLink /> renders with event props 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 5,
      "borderTopEndRadius": 20,
      "borderTopStartRadius": 20,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 40,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 20,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="ios-calendar"
    size={100}
    style={
      Object {
        "color": "black",
      }
    }
  />
  <Styled(Text)>
    Event
  </Styled(Text)>
</TouchableOpacity>
`;

exports[`<QuickLink /> renders with news props 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 5,
      "borderTopEndRadius": 20,
      "borderTopStartRadius": 20,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 40,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 20,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="md-paper"
    size={100}
    style={
      Object {
        "color": "black",
      }
    }
  />
  <Styled(Text)>
    News
  </Styled(Text)>
</TouchableOpacity>
`;

exports[`<QuickLink /> renders with profile props 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 5,
      "borderTopEndRadius": 20,
      "borderTopStartRadius": 20,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 40,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 20,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="ios-person"
    size={100}
    style={
      Object {
        "color": "black",
      }
    }
  />
  <Styled(Text)>
    Profile
  </Styled(Text)>
</TouchableOpacity>
`;
